rollup.config.js

```diff
@@ -1,44 +1,54 @@
+import { babel } from "@rollup/plugin-babel";
+import commonjs from "@rollup/plugin-commonjs";
```

Comment:
I'm not sure why we need this one?

Reply:
Because this is currently setup to use [@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env) with [`useBuiltIns: 'usage'`](https://babeljs.io/docs/en/babel-preset-env#usebuiltins-usage), it automatically pulls in any necessary polyfills from [core-js](https://github.com/zloirock/core-js). Since we want to include those polyfills in the distributed bundles, we need to use this plugin to convert them from CommonJS modules to ES6, so Rollup can include them. See: https://github.com/rollup/plugins/tree/master/packages/node-resolve#using-with-rollupplugin-commonjs

Reply:
I see, thanks. Does it only pull in polyfills that are actually needed or everything? Do you have any sense of how much this increases the size of the bundles?

Reply:
> Does it only pull in polyfills that are actually needed or everything?

It only pulls in the polyfills that are necessary for the target (node v12) based on the code actually used. In this case, I think it's these two small polyfills:

- https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.error.cause.js (used throughout)
- https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.reflect.to-string-tag.js (only used by `space-accessors.js`)

> Do you have any sense of how much this increases the size of the bundles?

It increases the bundle size by about 44% (non-minified) or 30% (minified):

Before:

```
Created bundle color.global.js: 107.94 kB → 34.67 kB (gzip)
Created bundle color.js: 104.35 kB → 34.23 kB (gzip)
Created bundle color.cjs: 104.42 kB → 34.27 kB (gzip)
Created bundle color.min.js: 41.59 kB → 16.48 kB (gzip)
Created bundle color.global.min.js: 41.6 kB → 16.52 kB (gzip)
Created bundle color.min.cjs: 41.66 kB → 16.5 kB (gzip)
```

After:

```
Created bundle color.global.js: 155.44 kB → 44.57 kB (gzip)
Created bundle color.js: 151.04 kB → 44.22 kB (gzip)
Created bundle color.cjs: 151.12 kB → 44.25 kB (gzip)
Created bundle color.min.js: 54.88 kB → 21.45 kB (gzip)
Created bundle color.global.min.js: 54.89 kB → 21.55 kB (gzip)
Created bundle color.min.cjs: 54.95 kB → 21.48 kB (gzip)
```

If you're concerned about that, I suppose another option would be to add new bundles that have support for older versions of Node, and not run everything through Babel?

Reply:
Hey, so sorry for the delay, I didn't get a notification and only saw you had responded when I manually visited the PR to see if there were any updates. Ouch, that's quite an increase. Yes, I think the way to go would be to have separate bundles. In that case we could even support somewhat older browsers too in those bundles, what do you think?
Is there a good naming convention for the transpiled bundles that we could follow?

Reply:
> Yes, I think the way to go would be to have separate bundles. In that case we could even support somewhat older browsers too in those bundles, what do you think? Is there a good naming convention for the transpiled bundles that we could follow?

@LeaVerou I agree, that seems like a good way to go. Are you thinking multiple new bundles with different levels of browser/node support? Or one "legacy" bundle that has support for older platforms?

I think I'd lean toward the latter, and probably I'd propose something like `color-legacy.js`? 🤷 

I'll do a quick update to try that out with Node 12 -- let me know if there are other platforms you think should be targeted (see: https://babeljs.io/docs/en/options#targets).

Reply:
(The [most recent commit](https://github.com/LeaVerou/color.js/pull/215/commits/fb3c8102e9bc2d25be453f364dede9f45536a26f) doesn't work -- I'll push a fixed version shortly.)

Reply:
@LeaVerou Adjusted in https://github.com/LeaVerou/color.js/pull/215/commits/862425c7466dbf1034f2015ef01e8446d80cc00f so that the original bundles are unchanged.